src/utils/utils/units.ts

```diff
@@ -63,3 +49,90 @@ export const getHumanizedSize = (size: string, bytesOption: 'withB' | 'withoutB'
 
   return humanizeBinaryBytes(baseValue);
 };
+
+const extractUnitFromQuantityString = (quantityString: string) =>
+  quantityString.match(UNIT_REGEX)?.[0];
+
+const extractNumberFromQuantityString = (quantityString: string) => {
+  const match = quantityString.match(NUMBER_REGEX);
+  return match ? parseFloat(match[0]) : null;
+};
+
+export const isBinaryUnit = (unit: string): unit is BinaryUnit =>
+  binaryUnitsOrdered.includes(BinaryUnit[unit]);
+
+export const isDecimalUnit = (unit: string): unit is DecimalUnit =>
+  decimalUnitsOrdered.includes(DecimalUnit[unit]);
+
+const isQuantityUnit = (unit: string): unit is QuantityUnit =>
+  isBinaryUnit(unit) || isDecimalUnit(unit);
+
+export const toQuantity = (quantityString: string, keepInitialUnit = true): Quantity => {
```

Comment:
1. IMHO we should unit test the utility functions in this file - they are likely to evolve in the future
2. please provide (perma)link to the format description - original not generated - https://github.com/kubevirt/kubevirt/blob/205d9455db56d5fcd42fb331122cfef358f19a69/vendor/k8s.io/apimachinery/pkg/api/resource/quantity.go#L33